### PR TITLE
Fix/simple-mint

### LINKED ISSUE #6343: Simple Minting on Kusama - Editions not working
### What happened?

Tried to mint and distribute some editions using rmrk/mint
it mints the NFT but, only 1 copy, even if i add up the number of editions
since it wont mint additional copies, no NFTS are sent

image

image

https://kusama.subscan.io/tx/0xa6b5ca30235a926b2b6f1b587c86eb9094b25e159cfe1cf7a1d3314f913785cc
https://kusama.subscan.io/tx/0x843ff176ff783044386e7bfcbefd4d49ba77023ac59525a638d1df9d701e8916

### Please reproduce in steps

RMRK -> Create - Simple

### Expected Behavior

Copies should be minted and sent to addresses

### What browsers are you seeing the problem on?

_No response_

### At which address did you encounter bug?

canary.kodadot.xyz

### Are you logged in?

None

### Which wallet you are using?

_No response_

### At which chain did you encounter bug?

_No response_

### Screenshots

^
| |

### Relevant log output

_No response_

### Payment link for reward

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #6388: Fix distribution on rmrk/mint
### What happened?

Distribution won't work as intended: 

@vikiival comment: https://github.com/kodadot/nft-gallery/pull/6380#issuecomment-1620315053

image

No NFTs got distributed

I'm also open this issue for discussion on how it should work and we could add a simple explanation on how the distribution works for better user understanding

### At which address did you encounter bug?

canary.kodadot.xyz

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Needs QA check

- @prury please review

## Context

- [x] Closes #6343
- [x] Closes #6388
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [ ] My fix has changed UI

### <samp>🤖 Generated by Copilot at b2f1e61</samp>

Fixed a type bug in the simple minting component and added support for minting NFTs under existing collections by providing the collection ID as a string. Modified the `getNftId` function in `scheme.ts` to handle both cases.

### <samp>🤖 Generated by Copilot at b2f1e61</samp>

> _To mint NFTs with more ease_
> _We added a new feature, please_
> _Check the `nft.collection`_
> _And its type detection_
> _And fix the `rmrkMint.max` keys_

### CODE REVIEW

File: components/rmrk/Create/SimpleMint.vue [hunk 1]
```diff
@@ -506,6 +506,7 @@ export default class SimpleMint extends mixins(
     this.isLoading = true
     this.status = 'loader.ipfs'
     const { accountId, version } = this
+    this.rmrkMint.max = Number(this.rmrkMint.max)
```
Comment (Jarsen136) [inline components/rmrk/Create/SimpleMint.vue:509]: Why convert its type two times in this file? Could we only keep one of them?

Comment (daiagi) [reply components/rmrk/Create/SimpleMint.vue:509]: done

File: components/rmrk/service/scheme.ts [hunk 2]
```diff
@@ -255,7 +255,9 @@ export const getNftId = (
   nft: Pick<NFT, 'blockNumber' | 'collection' | 'instance' | 'name' | 'sn'>,
   blocknumber?: string | number
 ): string => {
-  return `${blocknumber ? blocknumber + '-' : ''}${nft.collection.id}-${
+  const collectionId =
+    typeof nft.collection === 'string' ? nft.collection : nft.collection.id
+  return `${blocknumber ? blocknumber + '-' : ''}${collectionId}-${
```
Comment (vikiival) [inline components/rmrk/service/scheme.ts:260]: maybe good util for minimark?

**Comment by prury** [comment]
First test: 
image

tx: https://kusama.subscan.io/tx/0x24af384214522020f7262e4f6059e9b2e05bf338982f899b1d5539dfeae9e8b7
tx: https://kusama.subscan.io/tx/0xc43177b96299c8435b79e9fbcaf9afae80e43662987b4913792b2be23157fa10

4 editions successfully minted: 
image

**Distribution:** 

I've set the distribution to 75% via slider, as i understood 3 of 4 NFTs should have been sent to the addresses i had filled, but none got sent. 

@daiagi I'll open another issue to fix distribution

**Review by preschian — APPROVED** [review]

**Comment by daiagi** [comment]
successful mint with 2 editions:
https://kusama.subscan.io/extrinsic/18641451-5
/rmrk/collection/C4F63647002B182C0E-WOLF

successful mint with 2 editions + list for sell:
mint: https://kusama.subscan.io/extrinsic/18641561-3
list: https://kusama.subscan.io/extrinsic/18641565-3
/rmrk/collection/C4F63647002B182C0E-WOLF2

redirect after mint success (with and w/o listing for sell)

bugs for other issues:
1. no loader animation during the list for sell transaction
2.  > I've set the distribution to 75% via slider, as i understood 3 of 4 NFTs should have been sent to the addresses i had filled, but none got sent.

File: components/rmrk/service/scheme.ts (hunk 2 above)
Comment (daiagi) [reply components/rmrk/service/scheme.ts]: using `toNftId` instead of `getNftId`
 `toNftId` uses minimark utils

**Comment by vikiival** [comment]
> I've set the distribution to 75% via slider, as i understood 3 of 4 NFTs should have been sent to the addresses i had filled, but none got sent.

**I am checking the code that I have written around 06/2021**

How it works:

1. we parse all addressed
2. shuffle them via on-chain randomness
3. if distributon is set (e.g 75) - from 4 addr we take 3
4a. if you have less tokens than addresses let's say 2 we will take first 2 addresses
4b. if you have more tokens (10) we take 3 addresses as stated in step 3
5. transfer tx is done



If you see some logical error (it does not make sense) we can still change the algo

**Comment by prury** [comment]
> 



> > I've set the distribution to 75% via slider, as i understood 3 of 4 NFTs should have been sent to the addresses i had filled, but none got sent.
> 
> **I am checking the code that I have written around 06/2021**
> 
> How it works:
> 
>     1. we parse all addressed
> 
>     2. shuffle them via on-chain randomness
> 
>     3. if distributon is set (e.g 75) - from 4 addr we take 3
>        4a. if you have less tokens than addresses let's say 2 we will take first 2 addresses
>        4b. if you have more tokens (10) we take 3 addresses as stated in step 3
> 
>     4. transfer tx is done
> 
> 
> If you see some logical error (it does not make sense) we can still change the algo

oh, it works like a random draw thing then
the collection I've created had 4 nfts, 3 addresses and a 75% distribution percentage
at least 2 should have been transferred, but no one got an NFT

**Comment by vikiival** [comment]
> but no one got an NFT 

ah @daiagi I know
In: `@components/accounts/utils`
line 57 
you need to pass different version 💯

**Comment by daiagi** [comment]
> > but no one got an NFT
> 
> ah @daiagi I know In: `@components/accounts/utils` line 57 you need to pass different version 💯

@vikiival  I removed the version param as per `createInteraction` signature:

image

**Comment by vikiival** [comment]
>  I removed the version param as per createInteraction signature:


Oh therefore you need to import also createInteraction from minimark/v2 and if prefix is kusama then use v2

**Comment by daiagi** [comment]
>Oh therefore you need to import also createInteraction from minimark/v2 and if prefix is kusama then use v2

alrighty, to be continues tommorow

**Comment by daiagi** [comment]
>Oh therefore you need to import also createInteraction from minimark/v2 and if prefix is kusama then use v2

done
@prury  wanna give it a try?

**Review by vikiival — COMMENTED** [review]
Should be oki now

**Comment by prury** [comment]
> > Oh therefore you need to import also createInteraction from minimark/v2 and if prefix is kusama then use v2
> 
> done @prury wanna give it a try?

lemme try

**Comment by prury** [comment]
Seems that is working, let me do another one
number of editions: 5
number of addresses: 3
distribution percentage: 60%
60% of 3 = 1.8
1 NFT was successfully sent. 


collection:
https://deploy-preview-6380--koda-canary.netlify.app/rmrk/collection/7EA1DCF47E98A25067-FB?collectionId=7EA1DCF47E98A25067-FB

transferred NFT:
https://deploy-preview-6380--koda-canary.netlify.app/rmrk/gallery/18655081-7EA1DCF47E98A25067-FB-FUTURISTIC_BUILDING_1-0000000000000001


minting: 
https://kusama.subscan.io/tx/0x1a562ce19e6b645828bed07ffd2bace6dab91241633fe933924f6525ad88fa91
transfer: 
https://kusama.subscan.io/tx/0x29a6243a7d226ea4fffef1b2ecf716865166122ca5d805426785f41dbb0bed59

**Comment by prury** [comment]
trying now with less copies than addresses: 
image

Settings:
2 editions
listing for 0.4
4 addresses
100% distribution

collection:
https://deploy-preview-6380--koda-canary.netlify.app/rmrk/collection/7EA1DCF47E98A25067-IH?collectionId=7EA1DCF47E98A25067-IH

NFTs got sent successfully

https://kusama.subscan.io/tx/0x834d2c358595380661d4395f2c67f271d7066756c02691a7aa2f15fa0b2453e3
https://kusama.subscan.io/tx/0x9265ce9d0f536a5b8d805f0b2f0e25f86ddf3a2110bc12bc1f6e3be94f63cc4f

obs: listing was just a unused extra step because i was transferring them to other accounts

**Comment by prury** [comment]
last one: 
https://deploy-preview-6380--koda-canary.netlify.app/rmrk/collection/7EA1DCF47E98A25067-ANIME90?collectionId=7EA1DCF47E98A25067-ANIME90

settings:
4 addresses
6 editions
list for 0.3 ksm
distribution: 80%

3 got sent 3 got listed 

https://kusama.subscan.io/tx/0x63c6df3ba148c7a1b4f64c9648f35cefa1dfc0b82b64358f20d39c6d456cd6bc
https://kusama.subscan.io/tx/0x9256363daf71777212bd5734789a1cc7d42a1bc3f5456d9b8d939c8a5f4ddc30

**Comment by vikiival** [comment]
I got NFT 🥹

**Comment by daiagi** [comment]
> is it possible to change Editions to Copies like we did for statemine?

yes. done

**Comment by vikiival** [comment]
Sooo let's merge this

**Comment by vikiival** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 5.29 USD/DOT ~ 18.904 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x993cd44ec0d6ead6f671d789c2d21c7ecd7501e64726edd8e17fdf3566774d03](https://polkadot.subscan.io/extrinsic/0x993cd44ec0d6ead6f671d789c2d21c7ecd7501e64726edd8e17fdf3566774d03)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**